site/docs/advanced/flood.md

```diff
@@ -5,29 +5,119 @@ next: ./transformers.md
 
 # Scaling Up IV: Flood Limits
 
-Telegram restricts how many messages your bot can send per second, confer the [Bot FAQ](https://core.telegram.org/bots/faq#my-bot-is-hitting-limits-how-do-i-avoid-this).
-You should always make sure to stay below these limits, otherwise your bot gets rate limited.
-If you ignore these errors, your bot may eventually be banned.
+Telegram restricts how many messages your bot can send per second.
+This means that any API request you perform could error with status code 429 (Too Many Requests) and a `retry_after` header as specified [here](https://core.telegram.org/bots/api#responseparameters).
+This can happen anytime.
 
-## The Simple Solution
+There is only one correct way to handle such errors.
```

Comment:
```suggestion
There is only one correct way to handle that error:
```

Reply:
I don't like that it uses singular

Reply:
There’s only one error in the scope of flood waits: flood wait error. I also don’t like that it is plural.

If you don’t feel like accepting that one, how about this:

```suggestion
There is only one correct way to handle these situations:
```

Reply:
Love it